application/usecases/readDirectMessage.ts

```diff
@@ -0,0 +1,64 @@
+import { Message, Client, MessageEmbed, MessageActionRow, MessageButton } from "discord.js";
+import { ChannelSlug } from "../../types";
+import QuestionChatService from "./sendMessageToChannel/sendPerguntaToChannel";
+import DiscordEmbedService from "../../infrastructure/service/discordEmbedService";
+import ChatService from "../../domain/service/chatService";
+import DiscordChatService from "../../infrastructure/service/discordChatService";
+import ChannelResolver from "../../domain/service/channelResolver";
+
+const askedRecently = new Set();
+class DirectMessage {
+  constructor(private message: Message, private client: Client, private channelResolver: ChannelResolver) {}
+
+  async validate() {
+    if (askedRecently.has(this.message.author.id)) {
+      this.message.channel.send("Ainda não podes enviar outra pergunta. Tenta mais tarde.");
+    } else {
+      const chatService: ChatService = new DiscordChatService(this.client);
+      if (this.message.author.id === this.client.user?.id) {
+        return false;
+      }
+
+      if (
+        this.message.channel.type === "DM" &&
+        this.message.content.startsWith("!pergunta") &&
+        this.message.content.split(" ").length > 1 &&
+        this.message.content.length <= 1500
+      ) {
+        askedRecently.add(this.message.author.id);
+        setTimeout(() => {
+          askedRecently.delete(this.message.author.id);
+        }, 60000);
+        return true;
+      }
+      chatService.sendDM(this.message.author.id, "Por favor usa o seguinte formato:\n!pergunta <mensagem>");
+      return false;
+    }
+    return true;
+  }
```

Comment:
O que achas de uma abordagem mais deste género? Penso que fique mais explícito que tem que esperar os 60s para executar a ação seguinte.

```suggestion

        await this.timeout(60000);
        askedRecently.delete(this.message.author.id);
        
        return true;
      }
      chatService.sendDM(this.message.author.id, "Por favor usa o seguinte formato:\n!pergunta <mensagem>");
      return false;
    }
    return true;
  }
  
  function timeout(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
  }
```

Reply:
só ficou esta por fazer porque ainda não percebi bem, tenho que ver com mais tempo, entretanto tento resolver !